test/e2e/performance_test.go

```diff
@@ -0,0 +1,371 @@
+func setupPerformanceTestPodsConnection(data *TestData) error {
+	npSpec := networkv1.NetworkPolicySpec{
+		PodSelector: metav1.LabelSelector{MatchLabels: map[string]string{"app": performanceAppLabel}},
+		Ingress: []networkv1.NetworkPolicyIngressRule{
+			{
+				From: []networkv1.NetworkPolicyPeer{
+					{
+						PodSelector: &metav1.LabelSelector{
+							MatchLabels: map[string]string{"app": performanceAppLabel},
+						},
+					},
+				},
+			},
+		},
+		PolicyTypes: []networkv1.PolicyType{networkv1.PolicyTypeIngress},
+	}
+	np := &networkv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{Name: podsConnectionNetworkPolicyName},
+		Spec:       npSpec,
+	}
+	_, err := data.clientset.NetworkingV1().NetworkPolicies(testNamespace).Create(np)
+	return err
+}
+
+func generateWorkload(amount int) *networkv1.NetworkPolicy {
+	var ingressRules []networkv1.NetworkPolicyPeer
+	rndSrc := rand.NewSource(seed)
+	existed := make(map[string]struct{}) // ensure no duplicated cidrs
+	for len(ingressRules) < amount {
+		cidr := randCidr(rndSrc)
+		if _, ok := existed[cidr]; ok {
+			continue
+		} else {
+			existed[cidr] = struct{}{}
+		}
+		ingressRules = append(ingressRules, networkv1.NetworkPolicyPeer{IPBlock: &networkv1.IPBlock{CIDR: cidr}})
+	}
+	npSpec := networkv1.NetworkPolicySpec{
+		PodSelector: metav1.LabelSelector{MatchLabels: map[string]string{"app": performanceAppLabel}},
+		Ingress:     []networkv1.NetworkPolicyIngressRule{{From: ingressRules}},
+		PolicyTypes: []networkv1.PolicyType{networkv1.PolicyTypeIngress},
+	}
+	return &networkv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{Name: workloadNetworkPolicyName},
+		Spec:       npSpec,
+	}
+}
+
+func populateWorkload(np *networkv1.NetworkPolicy, data *TestData) error {
+	_, err := data.clientset.NetworkingV1().NetworkPolicies(testNamespace).Create(np)
+	return err
+}
+
+func setupPerformanceTestPods(data *TestData, b *testing.B) (nginxPodIP, abPodIP string) {
+	var err error
+	nginxPodIP, err = createPerformanceNginx(data, b)
+	if err != nil {
+		b.Fatalf("Error when waiting for IP assignment of nginx test pod: %v", err)
+	}
+
+	abPodIP, err = createPerformanceAB(data, b)
+	if err != nil {
+		b.Fatalf("Error when waiting for IP assignment of ab test pod: %v", err)
+	}
+	return nginxPodIP, abPodIP
+}
+
+func httpRequest(times int, workloadAmount int, data *TestData, b *testing.B) {
+	nginxPodIP, _ := setupPerformanceTestPods(data, b)
+	// enable pods connectivity policy first
+	if err := setupPerformanceTestPodsConnection(data); err != nil {
+		b.Fatalf("Error when adding network policy to set up connection between performance test pods")
+	}
+	b.Log("Populating performance test workloadAmount")
+	if err := populateWorkload(generateWorkload(workloadAmount), data); err != nil {
+		b.Fatalf("Error when populating workload: %v", err)
+	}
+
+	select { // wait workloads to be realized
```

Comment:
Please use one of the utility functions in https://godoc.org/k8s.io/apimachinery/pkg/util/wait for this. It will be more consistent with the rest of the test code and easier to read.

Reply:
Done.